### DISCUSSION TITLE
Blueprint Update

### DISCUSSION THREAD

**Discussion by vitalikkh** [Blackymas/NSPanel_HA_Blueprint#2258] [General]
I am sorry, this could very trivial questions but unfortunately I cannot find an answer by searching everything I can.
Current reported BP version is 4.3.1
After resent HA updates the HA complains about version mismatch saying the latest should be 4.3.11
So, do I REALLY have to update a BP to the latest 4.3.11 and if "yes" how do I do this?
 
Thank you,

- Vitaliy
-

**Comment by edwardtfn** [comment]
You must have all the 3 parts (Blueprint, ESPHome and TFT) with the same version, not necessarily the latest version.

Follow some useful links:

1. [How to update the Blueprint](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/howto.md#update-blueprint)
2. [How to install other versions](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/different_version.md)

Have I answered your question?

**Comment by vitalikkh** [comment]
Yes, Thank you very much!

⁣Get BlueMail for Android ​
